core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CloudflowFlinkPlugin.scala

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cloudflow.sbt
+
+import java.nio.charset.StandardCharsets._
+import java.util.Base64
+import java.util.zip.Deflater
+import java.io.ByteArrayOutputStream
+
+import com.typesafe.config._
+import sbt._
+import sbt.Keys._
+import sbtdocker._
+import sbtdocker.DockerKeys._
+import com.typesafe.sbt.packager.Keys._
+import com.typesafe.sbt.packager.archetypes._
+import spray.json._
+import JsonUtils._
+import cloudflow.sbt.CloudflowKeys._
+import cloudflow.blueprint._
+import cloudflow.blueprint.StreamletDescriptorFormat._
+import cloudflow.sbt.CloudflowKeys._
+import cloudflow.blueprint.StreamletDescriptor
+
+object CloudflowFlinkPlugin extends AutoPlugin {
+  val AppHome = "${app_home}"
+  val AppTargetDir: String = "/app"
+  val appTargetSubdir: String ⇒ String = dir ⇒ s"$AppTargetDir/$dir"
+  val AppJarsDir: String = "app-jars"
+  val DepJarsDir: String = "dep-jars"
+  val optAppDir = "/opt/cloudflow/"
+  final val CloudflowFlinkDockerBaseImage = "lightbend/flink:cloudflow-flink-1.9.2-scala-2.12"
+
+  override def requires = CommonSettingsAndTasksPlugin && StreamletScannerPlugin &&
+    JavaAppPackaging && sbtdocker.DockerPlugin
+
+  override def projectSettings = Seq(
+    libraryDependencies ++= Vector(
+      "com.lightbend.cloudflow" % "cloudflow-runner" % BuildInfo.version,
+      "com.lightbend.cloudflow" %% "cloudflow-flink" % BuildInfo.version,
+      "com.lightbend.cloudflow" %% "cloudflow-flink-testkit" % BuildInfo.version % "test"
+    ),
+    cloudflowDockerParentImage := CloudflowFlinkDockerBaseImage,
+
+    cloudflowFlinkDockerImageName := Def.task {
+      Some(DockerImageName((ThisProject / name).value.toLowerCase, (ThisProject / cloudflowBuildNumber).value.buildNumber))
```

Comment:
Can this be non-optional? (or what happens when it is none?)

Reply:
Not sure I understand. `DockerImageName` is always constructed as long as we have a project name and a build number. And `sbt.Task` takes an `Option[DockerImageName]`

Reply:
`sbt.Task` is `sbt.Task[T]`?

Reply:
You defined it as
```  val cloudflowFlinkDockerImageName = taskKey[Option[DockerImageName]]("The name of the Flink Docker image to publish.")```
But I wonder if this setting is even needed, when does the user change it / override it?

Reply:
@RayRoestenburg I was looking at `ImagePlugin` which does this with a key. But I agree we don't need to have it overridable. Any idea why we did that in `ImagePlugin` ?

Reply:
Not anymore, there must be some reason for it before, but if it is not needed now, let's not do it.